apps/files_sharing/lib/Controller/Share20OcsController.php

```diff
@@ -655,6 +689,31 @@ public function getShares() {
 		$path = $this->request->getParam('path', null);
 
 		$includeTags = $this->request->getParam('include_tags', false);
+		$shareTypes = $this->request->getParam('share_types', '');
+		if ($shareTypes === '') {
+			$shareTypes = [
+				Share::SHARE_TYPE_USER,
+				Share::SHARE_TYPE_GROUP,
+				Share::SHARE_TYPE_LINK,
+				Share::SHARE_TYPE_REMOTE,
+			];
+		} else {
+			$shareTypes = \explode(',', $shareTypes);
+		}
+
+		$requestedShareTypes = [
+			Share::SHARE_TYPE_USER => false,
+			Share::SHARE_TYPE_GROUP => false,
+			Share::SHARE_TYPE_LINK => false,
+			Share::SHARE_TYPE_REMOTE => false,
+		];
```

Comment:
IMO, we can simplify the adding share foreach logic with a control like the one below. 
```suggestion
		];
		if ($this->shareManager->outgoingServer2ServerSharesAllowed() === false) {
			\unset($requestedShareTypes[Share::SHARE_TYPE_REMOTE])
		}
```

Reply:
code adjusted